packages/edit-site/src/components/global-styles/font-library-modal/local-fonts.js

```diff
@@ -0,0 +1,181 @@
+		// Assuming that fromDataBuffer triggers onload event and returning a Promise
+		const onloadEvent = await new Promise(
+			( resolve ) => ( fontObj.onload = resolve )
+		);
+		const font = onloadEvent.detail.font;
+		const { name } = font.opentype.tables;
+		const fontName = name.get( 16 ) || name.get( 1 );
+		const isItalic = name.get( 2 ).toLowerCase().includes( 'italic' );
+		const fontWeight =
+			font.opentype.tables[ 'OS/2' ].usWeightClass || 'normal';
+		const isVariable = !! font.opentype.tables.fvar;
+		const weightAxis =
+			isVariable &&
+			font.opentype.tables.fvar.axes.find(
+				( { tag } ) => tag === 'wght'
+			);
+		const weightRange = weightAxis
+			? `${ weightAxis.minValue } ${ weightAxis.maxValue }`
+			: null;
+		return {
+			file: fontFile,
+			fontFamily: fontName,
+			fontStyle: isItalic ? 'italic' : 'normal',
+			fontWeight: weightRange || fontWeight,
+		}
+	}
+
+	const addFontFaceToBrowser = async  (fontFaceData ) => {
+		const data = await fontFaceData.file.arrayBuffer();
+		const newFont = new FontFace( fontFaceData.fontFamily, data, {
+			style: fontFaceData.fontStyle,
+			weight: fontFaceData.fontWeight,
+		} );
+		const loadedFace = await newFont.load();
+		document.fonts.add( loadedFace );
+	} 
+
+	/**
+	 * Creates the font family definition and sends it to the server
+	 *
+	 * @param {Array} fontFaces The font faces to be installed
+	 * @return {void}
+	 */
+	const handleInstall = async ( fontFaces ) => {
+		const formData = new FormData();
+		// Creates the fontFamilies array that will be sent to the server
+		const fontFamiliesObject = fontFaces.reduce(
+			( acc, { file, ...item }, i ) => {
+				if ( ! acc[ item.fontFamily ] ) {
+					acc[ item.fontFamily ] = {
+						name: item.fontFamily,
+						fontFamily: item.fontFamily,
+						slug: item.fontFamily
+							.replace( /\s+/g, '-' )
+							.toLowerCase(),
+						fontFace: [],
+					};
+				}
+				// Add the files to the formData
+				formData.append( `files${ i }`, file, file.name );
+				// Add the posted file id to the fontFace object
+				// This is needed to associate the fontFace with the file on the server
+				const face = { ...item, uploadedFile: `files${ i }` };
+				acc[ item.fontFamily ].fontFace.push( face );
+				return acc;
+			},
+			{}
+		);
+		const fontFamilies = Object.values( fontFamiliesObject );
+		// Adds the fontFamilies to the formData
+		formData.append( 'fontFamilies', JSON.stringify( fontFamilies ) );
+		await installFonts( formData );
+	};
+
+	return (
+		<>
+			<Text className="font-library-modal__subtitle">{ __( "Upload Fonts" ) }</Text>
+			<Spacer margin={ 2 } />
+			<DropZone onFilesDrop={ handleDropZone } />
+			<FormFileUpload
+				accept="font/*"
+				multiple={ true }
+				onChange={ onFilesUpload }
+				render={ ( { openFileDialog } ) => (
+					<div className="font-library-modal__upload-area" onClick={ openFileDialog }>
+						<span>{ __("Drag and drop you font files here.") }</span>
```

Comment:
```suggestion
						<span>{ __("Drag and drop your font files here.") }</span>
```
Small typo.

Alternatively, it would be nice to indicate that the area can be clicked to activate the upload dialog:
```suggestion
						<span>{ __("Click to upload, or drag and drop your font files here.") }</span>
```